Major rewrite with unit tests to align with GitHub web hooks format

nichtich opened the pull request:
Hi, thanks for the module! I refactored it to be covered with unit tests (see [travis](https://travis-ci.org/nichtich/Git-Hook-PostReceive) and [coveralls](https://coveralls.io/r/nichtich/Git-Hook-PostReceive), to align with [GitHub web hooks format](https://help.github.com/articles/post-receive-hooks) and to be more documented. Would you mind updating the module although the payload changes are not fully backwards-compatible? I can also take over the module if you are not interested anymore.

nichtich committed:
removed dependency on methods-invoker

nichtich committed:
removed dependency on Moose

nichtich committed:
require Perl 5.10 instead of 5.12

nichtich committed:
migrated to Dist::Zilla

nichtich committed:
enable travis-ci and coveralls.io

nichtich committed:
fixed and tested constructor

nichtich committed:
clarify that one can push multiple branches

nichtich committed:
basic unit test

nichtich committed:
return timestamp in XMLSchema format like GitHub does

nichtich committed:
fix whitespace in commit message

nichtich committed:
avoid git error message

nichtich committed:
modified payload format to align with GitHub

nichtich committed:
rewrite to extend payload format

nichtich committed:
see also

c9s commented:
Hi!

Thanks for the pull request. Any changes are welcomed, As to the API spec backward compatibility, we use a lot of
Of this, if you changed, please let me know and describe the changes, so that we can update our application.

Thanks!

Pedro

nichtich commented:
The payload is [described in documentation](https://github.com/nichtich/Git-Hook-PostReceive/blob/master/lib/Git/Hook/PostReceive.pm#L166). Payload changes to align with GitHub:
- `before` and `after` are not set to undef if they are "0000000000000000000000000000000000000000"
- renamed `new_head` to `created` and always supply this field with value 0 or 1
- renamed `delete` to `deleted` and always supply this field with value 0 or 1
- removed `ref_type` and return full branch (e.g. "refs/origin/master") instead of local name (e.g. "master") in `ref`
- renamed `date` to `timestamp` and return XMLSchema format (e.g. "2013-07-30T08:20:24+02:00") instead of DateTime::Format::DateParse
- additional fields `added`, `removed`, `modified` and `commiter`
- fixed whitespace bug in `message` field

See also the unit test for [a payload example](https://github.com/nichtich/Git-Hook-PostReceive/blob/master/t/run.t#L29).

Internal changes:
- moved to dist.zilla instead of Module::Install Makefile and shipit

One could write a function that maps the new payload format to the old format, but migrating to the GitHub-like format might be better for the future because the same format is also used elsewhere.

c9s commented:
yeah, agree. thanks for describing the details!

awesome
